### PR TITLE
update to latest AWS go SDK v2

### PR DESCRIPTION
**Pull request by hangxie**
These are the latest versions (as for now) that break backward compatibility so have to make code change https://github.com/aws/aws-sdk-go-v2/issues/2370

### CODE REVIEW

File: s3v2/s3.go [hunk 1]
```diff
@@ -395,8 +395,8 @@ func (s *S3File) openRead() error {
 
 	s.lock.Lock()
 	s.readOpened = true
-	if hoo.ContentLength != 0 {
-		s.fileSize = hoo.ContentLength
+	if hoo.ContentLength != nil {
```
Comment (lazareviczoran) [inline s3v2/s3.go:398]:
```suggestion
	if hoo.ContentLength != nil && *hoo.ContentLength != 0 {
```
I think just checking if it's not nil is not gonna provide exactly the same behavior as the current version

**Comment by adamwong246** [comment]
Just chiming in that our team is blocked on this. Anything we can do to help?

**Comment by lazareviczoran** [comment]
@xitongsys any chance you can review this PR soon?

Thanks in advance!

**Comment by MateuszKepczynski** [comment]
This update would also be useful to me. 😀

**Comment by fzerorubigd** [comment]
It seems that this PR is not getting merged any time soon, so you can use the replace directive for now like this in your go mod. 
```
replace github.com/xitongsys/parquet-go-source => github.com/hangxie/parquet-go-source v0.0.0-20231120172956-e0ff36b2d553

```

**Pull request merged**